db/rdb/src/main/java/it/unibz/inf/ontop/dbschema/impl/json/JsonUnionLens.java

```diff
@@ -0,0 +1,441 @@
+import it.unibz.inf.ontop.model.atom.DistinctVariableOnlyDataAtom;
+import it.unibz.inf.ontop.model.term.DBConstant;
+import it.unibz.inf.ontop.model.term.ImmutableTerm;
+import it.unibz.inf.ontop.model.term.TermFactory;
+import it.unibz.inf.ontop.model.term.Variable;
+import it.unibz.inf.ontop.model.type.DBTermType;
+import it.unibz.inf.ontop.substitution.ImmutableSubstitution;
+import it.unibz.inf.ontop.substitution.SubstitutionFactory;
+import it.unibz.inf.ontop.utils.ImmutableCollectors;
+import it.unibz.inf.ontop.utils.VariableGenerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+@JsonDeserialize(as = JsonUnionLens.class)
+public class JsonUnionLens extends JsonLens {
+    @Nonnull
+    public final List<List<String>> unionRelations;
+
+    @Nonnull
+    public final String provenanceColumn;
+
+    public final boolean makeDistinct;
+
+    protected static final Logger LOGGER = LoggerFactory.getLogger(JsonUnionLens.class);
+
+    @JsonCreator
+    public JsonUnionLens(@JsonProperty("name") List<String> name,
+                         @JsonProperty(value = "unionRelations", required = true) List<List<String>> unionRelations,
+                         @JsonProperty(value = "provenanceColumn") String provenanceColumn,
+                         @JsonProperty("makeDistinct") boolean makeDistinct,
+                         @JsonProperty("uniqueConstraints") UniqueConstraints uniqueConstraints,
+                         @JsonProperty("otherFunctionalDependencies") OtherFunctionalDependencies otherFunctionalDependencies,
+                         @JsonProperty("foreignKeys") ForeignKeys foreignKeys,
+                         @JsonProperty("nonNullConstraints") NonNullConstraints nonNullConstraints,
+                         @JsonProperty("iriSafeConstraints") IRISafeConstraints iriSafeConstraints) {
+        super(name, uniqueConstraints, otherFunctionalDependencies, foreignKeys, nonNullConstraints, iriSafeConstraints);
+        this.unionRelations = unionRelations;
+        this.provenanceColumn = provenanceColumn;
+        this.makeDistinct = makeDistinct;
+    }
+
+    @Override
+    public Lens createViewDefinition(DBParameters dbParameters, MetadataLookup parentCacheMetadataLookup)
+            throws MetadataExtractionException {
+
+        QuotedIDFactory quotedIDFactory = dbParameters.getQuotedIDFactory();
+        RelationID relationId = quotedIDFactory.createRelationID(name.toArray(new String[0]));
+
+        IQ iq = createIQ(relationId, dbParameters, parentCacheMetadataLookup);
+
+        int maxParentLevel = extractMaxParentLevel(iq, dbParameters.getCoreSingletons());
+
+        if (maxParentLevel > 0)
```

Comment:
@DRovara , please remove this test and the warning (not relevant for union lenses)

Reply:
Done!